### PR TITLE
feat: port to micro 2.0.1+

### PR DESCRIPTION
**Pull request by joshuarli**
Since micro 2 is in the works, which includes a breaking overhaul of how it exposes parts of itself to plugins, this plugin needs to be updated accordingly. So I've started porting + refactoring it.

As of 53c35e0 there's still some work (like implementing charset setting) and manual testing/verification left to do, so marking this as WIP. Reaching feature parity is also blocked by upstream micro as of the time of writing, since it doesn't let plugins write to the log buffer.

Once micro 2.0.0 reaches a stable release, we'll need to bump the version to 1.0.0 before merging, and also it would be nice to make sure that simply updating master won't break micro <2's plugin installer (I haven't looked into exactly how it works, but I hope it doesn't just blindly download from master).

cc @zyedidia

### CODE REVIEW

**Comment by joshuarli** [comment]
Oh yeah @10sr, I suppose once micro 2.0.0 is out and this is merged, could we move this repo under https://github.com/editorconfig?

**Comment by 10sr** [comment]
Thanks! I really appreciate your continuous effort on this project.

> it would be nice to make sure that simply updating master won't break micro <2's plugin installer (I haven't looked into exactly how it works, but I hope it doesn't just blindly download from master).

Plugin informaton of micro<2 is defined here https://github.com/micro-editor/plugin-channel/blob/master/channel.json#L6: it download repo.json of master branch 😖 
So we likely have to leave this file (or create new branch for micro<2 ?).

> I suppose once micro 2.0.0 is out and this is merged, could we move this repo under https://github.com/editorconfig?

Sounds nice! 👍

**Comment by joshuarli** [comment]
Ah, thanks for verifying.

I suppose we'll have to update master's readme with special instructions to install the 2.0-compatible version. That'd be like, clone this to `plug` and then check out this branch.

Or, and I personally like this approach better, we can just archive this repository and keep it around for micro < 2 users, and start a new 2.0-compatible repository in the editorconfig organization with this branch. In this case, I'll leave it up to you to create that repo and grant me collaborator.

**Comment by zyedidia** [comment]
It should also be possible to modify the plugin channel to point to

https://raw.githubusercontent.com/10sr/editorconfig-micro/848fc839cad565c61d96b99f38121dda0f485607/repo.json

instead of the master branch.

**Comment by joshuarli** [comment]
@zyedidia Hmm, that might work, although actually now that I think about it, the easiest way would be to just keep the repo.json on master, assuming micro < 2 uses `"Url": "https://github.com/10sr/editorconfig-micro/archive/v0.2.3.zip",`. I'll restore repo.json and channel.json in this branch.

**Comment by zyedidia** [comment]
It's cleaner in my opinion to point to an older commit hash. It's not great to be keeping around cruft like the `repo.json` and `channel.json` when it's not necessary (looks like the `channel.json` will also have to be modified to point off of master).

**Comment by joshuarli** [comment]
Cool, I'll send a PR your way soon then.

**Comment by joshuarli** [comment]
> It's cleaner in my opinion to point to an older commit hash. It's not great to be keeping around cruft like the `repo.json` and `channel.json` when it's not necessary (looks like the `channel.json` will also have to be modified to point off of master).

For posterity, the plugin manager has since been brought back.

**Comment by joshuarli** [comment]
Some remaining TODOs:

    - wait until OpenLogBuf lands in micro
    - verify that micro added support for more charsets, and adapt accordingly
    - recommend that ftoptions default plugin be disabled, since it could be redundant depending on your configuration
	- verify that help/editorconfig.md is accurate (e.g. the custom keybinding settings syntax works)
    - test that the verbose config works (the editorconfigverbose in settings.json)

**Comment by zyedidia** [comment]
The OpenLogBuf function has been provided (`bp:OpenLogBuf()` will open the log buffer as a horizontal split with `bp`, or do nothing if `bp` is a log buffer).

**Comment by joshuarli** [comment]
@10sr Alright, this seems to work well with micro 2.0.1. I'll merge with your approval after you review.

File: help/editorconfig.md
Comment (joshuarli) [inline help/editorconfig.md:19]: @10sr In case you were wondering, these two other ways of executing the plugin don't work anymore. According to debug logs, it is being run, but things like rmtrailingws aren't actually modifying the buffer for some reason. But I opted to just remove them from documentation since I feel like no one uses them. `"editorconfig.getApplyProperties"` doesn't turn up in GitHub-wide search in peoples' dotfiles, and I can't imagine anyone typing out `editorconfig` when they can just instead save the file.

**Comment by joshuarli** [comment]
@10sr Does your 👀 react mean you've seen it and don't have anything to say (approving), or do you still need to look at this?

**Comment by 10sr** [comment]
Sorry I was busy with work recently... 😫 
I'll check this week!

File: help/editorconfig.md
Comment (10sr) [reply help/editorconfig.md:19]: This was defined because originally this plugin did not run getApplyProperties on save.
Now it does, so OK to remove this command 👍

**Review by 10sr — APPROVED** [review]
Awesome! 🚀

**Comment by 10sr** [comment]
Opened an issue of transferring this repository https://github.com/editorconfig/editorconfig/issues/419 🙌

**Comment by joshuarli** [comment]
> Opened an issue of transferring this repository [editorconfig/editorconfig#419](https://github.com/editorconfig/editorconfig/issues/419) raised_hands

Awesome, thanks!

**Pull request merged**
